### PR TITLE
[ReactiveElement] Throws rather than warns in dev mode for class fields that shadow reactive properties

### LINKED ISSUE #2104: [ReactiveElement] Make shadowed properties throw instead of warn in DEV_MODE
The warning is likely not enough since your element is really in a broken state if this happens. Instead we should throw.

### PR DESCRIPTION
**Pull request by sorvell**
Fixes #2104. The element is in a broken state in this case, so rather than warn, we throw.

### CODE REVIEW

**Review by kevinpschaaf — APPROVED** [review]
LGTM

File: packages/reactive-element/src/reactive-element.ts [hunk 1]
```diff
@@ -1092,16 +1092,13 @@ export abstract class ReactiveElement
           }
         );
         if (shadowedProperties.length) {
-          issueWarning(
+          throw new Error(
```
Comment (justinfagnani) [inline packages/reactive-element/src/reactive-element.ts:1095]: @aomarks what should we do for errors now that `issueWarning` takes a warning code?

**Review by justinfagnani — APPROVED** [review]
Code looks good, just a question about using the new message URLs with errors.

File: packages/reactive-element/src/reactive-element.ts (hunk 1 above)
Comment (kevinpschaaf) [reply packages/reactive-element/src/reactive-element.ts:1095]: The error can just include the full URL; don't need to go through issueWarning, right?

Comment (aomarks) [reply packages/reactive-element/src/reactive-element.ts:1095]: Yeah that seems fine, just put the code URL directly.

Comment (kevinpschaaf) [reply packages/reactive-element/src/reactive-element.ts:1098]: Done

**Pull request merged**